### ISSUE TITLE
Failed to get proxy for DBus daemon (org.freedesktop.DBus): Could not connect: No such file or directory

### ISSUE THREAD

**Issue by srinivasahk** [tpm2-software/tpm2-tss#1429]
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.2 LTS
Release:        18.04.

I m new to TPM2.0. I installed tpm2-tss, tpm2-tools, tpm2-abrmd & ibmtpm. When I try to check: 

/usr/local/sbin/tpm2-abrmd --tcti=mssim

** (tpm2-abrmd:14282): WARNING **: 20:56:18.557: Failed to get proxy for DBus daemon (org.freedesktop.DBus): Could not connect: No such file or directory

tpm0/tpmrm0 dir not created.

**Comment by tstruk** [comment]
Please follow the steps described in https://github.com/tpm2-software/tpm2-abrmd/blob/master/INSTALL.md

**Comment by srinivasahk** [comment]
Thanks for the reply. I followed all the steps but tcti & /dev/tpm0 is not created.  I installed dbus from get as well as from git source but the problem is not resolved.  

sri@srinivas:/mnt/e/TPM2.0/tpm2-abrmd/src$ sudo apt list --installed | grep autoconf
autoconf/bionic,now 2.69-11 all [installed]
autoconf-archive/bionic,now 20170928-2 all [installed]
sri@srinivas:/mnt/e/TPM2.0/tpm2-abrmd/src$ sudo apt list --installed | grep automake
automake/bionic,now 1:1.15.1-3ubuntu2 all [installed]
sri@srinivas:/mnt/e/TPM2.0/tpm2-abrmd/src$ sudo apt list --installed | grep libtool
libtool/bionic,now 2.4.6-2 all [installed]
sri@srinivas:/mnt/e/TPM2.0/tpm2-abrmd/src$ sudo apt list --installed | grep gcc
gcc/bionic-updates,now 4:7.4.0-1ubuntu2.2 amd64 [installed]
gcc-7/bionic-updates,now 7.4.0-1ubuntu1~18.04 amd64 [installed,automatic]
gcc-7-base/bionic-updates,now 7.4.0-1ubuntu1~18.04 amd64 [installed,automatic]
gcc-8-base/bionic-updates,now 8.3.0-6ubuntu1~18.04 amd64 [installed]
libgcc-7-dev/bionic-updates,now 7.4.0-1ubuntu1~18.04 amd64 [installed,automatic]
libgcc1/bionic-updates,now 1:8.3.0-6ubuntu1~18.04 amd64 [installed]
sri@srinivas:/mnt/e/TPM2.0/tpm2-abrmd/src$ sudo apt list --installed | grep pkg-config
pkg-config/bionic,now 0.29.1-0ubuntu2 amd64 [installed]
sri@srinivas:/mnt/e/TPM2.0/tpm2-abrmd/src$ sudo apt list --installed | grep glib
gir1.2-glib-2.0/bionic,now 1.56.1-1 amd64 [installed]
libdbus-glib-1-2/bionic,now 0.110-2 amd64 [installed,automatic]
libdbus-glib-1-dev/bionic,now 0.110-2 amd64 [installed]
libglib2.0-0/bionic-updates,now 2.56.4-0ubuntu0.18.04.2 amd64 [installed]
libglib2.0-bin/bionic-updates,now 2.56.4-0ubuntu0.18.04.2 amd64 [installed,automatic]
libglib2.0-data/bionic-updates,now 2.56.4-0ubuntu0.18.04.2 all [installed]
libglib2.0-dev/bionic-updates,now 2.56.4-0ubuntu0.18.04.2 amd64 [installed,automatic]
libglib2.0-dev-bin/bionic-updates,now 2.56.4-0ubuntu0.18.04.2 amd64 [installed,automatic]


sri@srinivas:/mnt/e/TPM2.0/tpm2-abrmd/src$ sudo apt list --installed | grep  libsapi



sri@srinivas:/mnt/e/TPM2.0/tpm2-abrmd/src$ sudo apt list --installed | grep  tcti


UT getting passed but Integration test for tcti is failing.. 

PASS: test/unit/esys-crypto
PASS: test/unit/esys-default-tcti
ERROR: test/integration/sapi-nv-readwrite.int
ERROR: test/integration/sapi-nv-policy-locality.int
ERROR: test/integration/sapi-asymmetric-encrypt-decrypt.int
ERROR: test/integration/sapi-create-keyedhash-sha1-hmac.int
ERROR: test/integration/sapi-primary-rsa-2K-aes128cfb.int
ERROR: test/integration/sapi-encrypt-decrypt.int
ERROR: test/integration/sapi-encrypt-decrypt-2.int
ERROR: test/integration/sapi-evict-ctrl.int
ERROR: test/integration/sapi-get-random.int
ERROR: test/integration/sapi-stir-random.int
ERROR: test/integration/sapi-hierarchy-change-auth.int
ERROR: test/integration/sapi-abi-version.int

**Comment by srinivasahk** [comment]
ERROR:tcti:src/tss2-tcti/tcti-device.c:399:Tss2_Tcti_Device_Init() Failed to open device file /dev/tpm0: No such file or directory

** (tpm2-abrmd:48): WARNING **: 21:20:49.572: Failed to get proxy for DBus daemon (org.freedesktop.DBus): Could not connect: No such file or directory

** (tpm2-abrmd:48): WARNING **: 21:20:49.573: failed to initialize device TCTI context: 0xa000a

** (tpm2-abrmd:48): CRITICAL **: 21:20:49.575: Failed to acquire DBus name com.intel.tss2.Tabrmd. UID 1000 must be allowed to "own" this name. Check DBus config and check that this is running as user tss or root.

** (tpm2-abrmd:48): CRITICAL **: 21:20:49.576: init_thread_func: failed to create TCTI with name "libtss2-tcti-device.so.0" and conf "(null)"

**Comment by tstruk** [comment]
On 5/25/19 1:16 AM, srinivasahk wrote:
> Thanks for the reply. I followed all the steps but tcti & /dev/tpm0 is not created. I installed dbus from get as well as from git source but the problem is not resolved.

It means that the kernel driver is disabled or you are using some very old kernel.

-- 
Tadeusz

**Comment by srinivasahk** [comment]
thanks tstruk.  After exploring few options, it seems, cant run software ubuntu on top of windows. It will work with dualboot.
I have bought tpm2  hw & started working on that.
Closing the case.

**Comment by srinivasahk** [comment]
Hi,
I was running tpm s/w on linux  & was in assumption tool is responding to commands but it wasnt the case. I was getting same response from tpm command even after closing tpm_server/abrmd. 


I tried to find related topic but not able to find any. Is there any document where I can:
1) if s/w will run if the laptop doesnt have tpm. (I tried this option, /dev/tpm*  not created)
2) In linux if tpm2 present, tpm2 is giving response instead of simulator. 

Thanks,
Srinivas.


    On Tuesday, 28 May 2019, 22:48:15 GMT+5:30, Tadeusz Struk <notifications@github.com> wrote:  
 
 On 5/25/19 1:16 AM, srinivasahk wrote:
> Thanks for the reply. I followed all the steps but tcti & /dev/tpm0 is not created. I installed dbus from get as well as from git source but the problem is not resolved.

It means that the kernel driver is disabled or you are using some very old kernel.

-- 
Tadeusz
